src/hardware/index.ts

```diff
@@ -136,20 +136,25 @@ export const getSignerFromRemoteData = (signer) => {
   };
 };
 
-export const getSignerDescription = (signer: Signer) => {
-  if (signer) {
-    if (signer.extraData?.givenName || signer.extraData?.familyName) {
-      const fullName = `${signer.extraData?.givenName || ''} ${
-        signer.extraData?.familyName || ''
-      }`.trim();
-      return fullName || 'Unknown Contact';
-    }
-    if (signer.signerDescription) {
-      return signer.signerDescription;
-    }
-    return `Added ${moment(signer.addedOn).calendar().toLowerCase()}`;
+export const getSignerDescription = (signer?: Signer) => {
+  const fullName = `${signer?.extraData?.givenName || ''} ${
+    signer?.extraData?.familyName || ''
+  }`.trim();
+
+  if (fullName) {
+    return fullName;
+  }
+
+  if (signer?.signerDescription) {
+    return signer.signerDescription;
+  }
+
+  if (signer?.type === SignerType.MY_KEEPER) {
+    const instanceNumber = signer?.extraData?.instanceNumber;
+    return instanceNumber !== undefined ? numberToOrdinal(instanceNumber) : 'Unknown Instance';
```

Comment:
It normally shouldn't happen, but if it does we probably don't want to show "Unknown Instance" to the user, maybe something less obsucre and more generic like "Mobile Key"? Or just continue to the final return of the added on time.